molgenis/molgenis-emx2
# feat(webapi): RDF API now supports HEAD calls similar to FDP API + added missing convience API calls

**IMPORTANT: This PR is build on https://github.com/molgenis/molgenis-emx2/pull/4126 as otherwise the tests failed no matter what. Until that PR is merged, this PR will remain a draft!**

What are the main changes you did:
- correct `content-type` is now returned for HEAD requests, similar to behaviour of `/api/fdp`:
https://github.com/molgenis/molgenis-emx2/blob/d5ac742d991409d2fa0f6d4f7ec621dd4fd085e5/backend/molgenis-emx2-webapi/src/main/java/org/molgenis/emx2/web/FAIRDataPointApi.java#L38-L39
https://github.com/molgenis/molgenis-emx2/blob/d5ac742d991409d2fa0f6d4f7ec621dd4fd085e5/backend/molgenis-emx2-webapi/src/main/java/org/molgenis/emx2/web/FAIRDataPointApi.java#L69-L71
- Fixed convenience APIs missing API points for row/column (f.e. `curl http://localhost:8080/pet%20store/api/jsonld/Pet` did nothing while `curl http://localhost:8080/pet%20store/api/rdf/Pet` worked)
- Reduced testing code duplicity.

how to test:
Should be covered already by unit-tests, but in case of manual testing:

| command line | tests | expected content-type |
| --- | --- | --- |
| `curl --head http://localhost:8080/pet%20store/api/rdf` | HEAD | text/turtle |
| `curl --head http://localhost:8080/pet%20store/api/ttl` | HEAD | text/turtle |
| `curl --head http://localhost:8080/pet%20store/api/jsonld` | HEAD | application/ld+json |
| `curl -H "Accept: application/ld+json" --head http://localhost:8080/pet%20store/api/rdf` | HEAD | application/ld+json |
| `curl -v http://localhost:8080/pet%20store/api/rdf \| grep "^>"` | GET | text/turtle |
| `curl -v http://localhost:8080/pet%20store/api/jsonld \| grep "^>"` | GET | application/ld+json |
| `curl -H "Accept: application/ld+json" -v http://localhost:8080/pet%20store/api/rdf \| grep "^>"` | GET | application/ld+json |

todo:
